src/fountain.rs

```diff
@@ -599,13 +601,26 @@ impl Part {
     }
 }
 
+/// Calculates the quotient of `a` and `b`, rounding the results towards
+/// positive infinity.
+///
+/// Note: there's an implementation on the `usize` type of this function,
+/// but it's not stable yet.
+#[must_use]
+fn div_ceil(a: usize, b: usize) -> usize {
+    let d = a / b;
+    let r = a % b;
+    if r > 0 && b > 0 {
+        d + 1
+    } else {
+        d
+    }
+}
+
 #[must_use]
-#[allow(clippy::cast_possible_truncation)]
-#[allow(clippy::cast_precision_loss)]
-#[allow(clippy::cast_sign_loss)]
 pub(crate) fn fragment_length(data_length: usize, max_fragment_length: usize) -> usize {
-    let fragment_count = data_length / max_fragment_length + 1;
```

Comment:
Interesting, if I look at the code, then the computation of `fragment_count` was actually wrong (off by +1) in case  `data_length` is divisible by `max_fragment_length`. Seems not to have been covered by the test suite.